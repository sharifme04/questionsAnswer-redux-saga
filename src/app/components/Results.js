import React from "react";
import PropTypes from "prop-types";

const Results = ({
  history,
  selectedIncorrect,
  selectedCorrect,
  updateResults
}) => {
  let correctAns =
    selectedCorrect &&
    selectedIncorrect &&
    selectedCorrect.length - selectedIncorrect.length;
  return (
    <div>
      <h2>Results</h2>
      <hr />
      <div className="card text-center">
        <div className="card-header">Your Score</div>
        <div className="card-body">
          {selectedCorrect && selectedCorrect.length ? (
            <div>
              <h3 className="card-title">
                {correctAns && correctAns}/
                {updateResults && updateResults.length}
              </h3>
              <p className="card-text mb-4">Congratulation!!!!</p>
            </div>
          ) : (
            <h5 className="card-title">
              You did not select any answer. Please Go back and answer all
              questions
            </h5>
          )}
          <button
            type="button"
            className="btn btn-primary"
            onClick={() => history.push("/")}
          >
            Back
          </button>
        </div>
        <div className="card-footer text-muted">Deevio GmbH</div>
      </div>
    </div>
  );
};
Results.propTypes = {
  selectedIncorrect: PropTypes.array,
  selectedCorrect: PropTypes.array,
  updateResults: PropTypes.array
};

export default Results;
