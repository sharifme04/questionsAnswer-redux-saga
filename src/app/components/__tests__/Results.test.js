import React from "react";
import { shallow } from "enzyme";
import Results from "../Results";

const fakeSelectedCorrect = [
  { 2: "January 25, 1945" },
  { 0: "Jerk" },
  { 1: "True" }
];

const fakeSelectedIncorrect = [
  { 2: "December 16, 1944" },
  { 0: "Shift" },
  { 1: "False" }
];

const fakeUpdateResults = [
  {
    category: "Entertainment: Film",
    type: "multiple",
    difficulty: "easy",
    question:
      "Who played Deputy Marshal Samuel Gerard in the 1993 film &quot;The Fugitive&quot;?",
    correct_answer: "Tommy Lee Jones",
    incorrect_answers: ["Harrison Ford", "Harvey Keitel", "Martin Landau"],
    allAnswers: [
      "Harrison Ford",
      "Harvey Keitel",
      "Martin Landau",
      "Tommy Lee Jones"
    ]
  }
];

describe("<Results />", () => {
  test("render", () => {
    const component = shallow(
      <Results
        updateResults={fakeUpdateResults}
        selectedCorrect={fakeSelectedCorrect}
        selectedIncorrect={fakeSelectedIncorrect}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("return the default when there is no data to map through", () => {
    const component = shallow(<Results />);
    expect(component).toMatchSnapshot();
  });

  test("does not break without value", () => {
    const component = shallow(<Results />);
    expect(component.find("button")).toHaveLength(1);
  });

  test("does not break with an empty array", () => {
    const component = shallow(
      <Results updateResults={[]} selectedCorrect={[]} selectedIncorrect={[]} />
    );
    expect(component.find("button")).toHaveLength(1);
  });
});
