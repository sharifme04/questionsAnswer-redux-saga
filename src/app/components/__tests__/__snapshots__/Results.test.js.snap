// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> render 1`] = `
<div>
  <h2>
    Results
  </h2>
  <hr />
  <div
    className="card text-center"
  >
    <div
      className="card-header"
    >
      Your Score
    </div>
    <div
      className="card-body"
    >
      <div>
        <h3
          className="card-title"
        >
          0
          /
          1
        </h3>
        <p
          className="card-text mb-4"
        >
          Congratulation!!!!
        </p>
      </div>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
    <div
      className="card-footer text-muted"
    >
      Deevio GmbH
    </div>
  </div>
</div>
`;

exports[`<Results /> return the default when there is no data to map through 1`] = `
<div>
  <h2>
    Results
  </h2>
  <hr />
  <div
    className="card text-center"
  >
    <div
      className="card-header"
    >
      Your Score
    </div>
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        You did not select any answer. Please Go back and answer all questions
      </h5>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
    <div
      className="card-footer text-muted"
    >
      Deevio GmbH
    </div>
  </div>
</div>
`;
